src/main/java/mate/project/service/UserServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package mate.project.service;
+
+import lombok.RequiredArgsConstructor;
+import mate.project.dto.UserRegistrationRequestDto;
+import mate.project.dto.UserResponseDto;
+import mate.project.exception.RegistrationException;
+import mate.project.mapper.UserMapper;
+import mate.project.model.User;
+import mate.project.repository.user.UserRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
```

Comment:
Consider not using checked exceptions for business logic, use unchecked/runtime exceptions instead.